lnbits/walletsfiat/base.py

```diff
@@ -0,0 +1,148 @@
+    def pending(self) -> bool:
+        return self.ok is None
+
+    @property
+    def failed(self) -> bool:
+        return self.ok is False
+
+
+class FiatPaymentResponse(NamedTuple):
+    # when ok is None it means we don't know if this succeeded
+    ok: bool | None = None
+    checking_id: str | None = None  # payment_hash, rcp_id
+    fee: float | None = None
+    error_message: str | None = None
+
+    @property
+    def success(self) -> bool:
+        return self.ok is True
+
+    @property
+    def pending(self) -> bool:
+        return self.ok is None
+
+    @property
+    def failed(self) -> bool:
+        return self.ok is False
+
+
+class FiatPaymentStatus(NamedTuple):
+    paid: bool | None = None
+    fee: float | None = None  # todo: what fee is this?
+
+    @property
+    def success(self) -> bool:
+        return self.paid is True
+
+    @property
+    def pending(self) -> bool:
+        return self.paid is not True
+
+    @property
+    def failed(self) -> bool:
+        return self.paid is False
+
+    def __str__(self) -> str:
+        if self.success:
+            return "success"
+        if self.failed:
+            return "failed"
+        return "pending"
+
+
+class FiatPaymentSuccessStatus(FiatPaymentStatus):
+    paid = True
+
+
+class FiatPaymentFailedStatus(FiatPaymentStatus):
+    paid = False
+
+
+class FiatPaymentPendingStatus(FiatPaymentStatus):
+    paid = None
+
+
+class FiatWallet(ABC):
+    @abstractmethod
+    async def cleanup(self):
+        pass
+
+    @abstractmethod
+    def status(
+        self, only_check_settings: bool | None = False
+    ) -> Coroutine[None, None, FiatStatusResponse]:
+        pass
+
+    @abstractmethod
+    def create_invoice(
+        self,
+        amount: float,
+        payment_hash: str,
+        currency: str,
+        memo: str | None = None,
+        **kwargs,
+    ) -> Coroutine[None, None, FiatInvoiceResponse]:
+        pass
+
+    @abstractmethod
+    def pay_invoice(
+        self,
+        payment_request: str,
+    ) -> Coroutine[None, None, FiatPaymentResponse]:
+        pass
+
+    @abstractmethod
+    def get_invoice_status(
+        self, checking_id: str
+    ) -> Coroutine[None, None, FiatPaymentStatus]:
+        pass
+
+    @abstractmethod
+    def get_payment_status(
+        self, checking_id: str
+    ) -> Coroutine[None, None, FiatPaymentStatus]:
+        pass
+
+    async def paid_invoices_stream(
+        self,
+    ) -> AsyncGenerator[str, None]:
+        yield ""
+
+    def normalize_endpoint(self, endpoint: str, add_proto=True) -> str:
+        endpoint = endpoint[:-1] if endpoint.endswith("/") else endpoint
+        if add_proto:
+            if endpoint.startswith("ws://") or endpoint.startswith("wss://"):
+                return endpoint
+            endpoint = (
+                f"https://{endpoint}" if not endpoint.startswith("http") else endpoint
+            )
+        return endpoint
+
+
+class UnsupportedError(Exception):
+    pass
```

Comment:
lets move this to `exceptions.py` and import from there, it is also in `wallets/base.py`

Reply:
done